tasks/determine-image-tag-task.yaml

```diff
@@ -119,9 +129,24 @@ spec:
           return
         fi
 
+        if [[ "${stackrox_repo}" == "yes" ]]; then
+          log "Using sanitized Makefile output ${tag_from_makefile_sanitized} for the tag."
+          echo "${tag_from_makefile_sanitized}"
+          return
+        fi
```

Comment:
We don't need to sanitize versions in Collector and Scanner repos because tags/versions produced during builds there are _internal_, _development_ versions (e.g. `3.20.x-80-g5afdaf059a` and `2.35.x-57-ga9da3070bd`) and they should match what we have in `COLLECTOR_VERSION` and `SCANNER_VERSION` files.

What Fast Stream users will work with are _release_ images which are retagged in StackRox to the unified main version, i.e. `4.7.0-551-g1ca8d0d66e-fast`.

Reply:
optionally add a comment with

> We don't need to and should not sanitize versions in Collector and Scanner repos because tags/versions produced during builds there are internal, development versions (e.g. 3.20.x-80-g5afdaf059a and 2.35.x-57-ga9da3070bd) and they should match what we have in COLLECTOR_VERSION and SCANNER_VERSION files.
> What Fast Stream users will work with are release images which are retagged in StackRox to the unified main version, i.e. 4.7.0-551-g1ca8d0d66e-fast.

Reply:
Will merge as-is.

Reply:
GitHub did not refresh the thread. Will Include the comment.